src/controllers/user/controller.ts

```diff
@@ -14,112 +20,137 @@ class UserController {
         return UserController.instance;
     }
 
-    login(req: IRequest, res: Response, next: NextFunction) {
-    try{ const { email, password } = req.body;
-        userModel.findOne({ email: email }, (err, docs) => {
-            if (docs) {
-                if (password === docs.password) {
-                    const token = jwt.sign({ docs }, 'qwertyuiopasdfghjklzxcvbnm123456');
-                    res.send({
-                        data: token,
-                        message: 'LoggedIN',
-                        status: 200
-                    })
+    public async login(req: IRequest, res: Response, next: NextFunction) {
+        const { email, password } = req.body;
+        const user = new UserRepository();
+        await user.get({ email })
+            .then((userData) => {
+                if (userData === null) {
+                    next({
+                        message: 'User does Not exist',
+                        error: 404,
+                    });
                 }
-                else {
+
+                if (password !== req.body.password) {
                     next({
-                        error: 'Not found',
-                        status: 404,
-                        message: 'Password not Mtached with DB'
-                    })
+                        message: 'Password is Invalid',
+                        err: 401,
+
+                    });
                 }
-            }
-            else {
-                next({
-                    error: 'Not found',
-                    status: 404,
-                    message: 'Email Not Exist in DB'
-                })
-            };
-        });
-    }
-    catch ( err ) {
-        res.send( err );
+                const expDate = new Date();
+                const payLoad = {
+                    name: 'sumit',
+                    iss: new Date(),
+                    exp: expDate.setDate(expDate.getDate() + 7),
+                    email: "sumit.upadhyay@successive.tech",
+                    role: 'trainee'
+                }
```

Comment:
Use dynamic payload.

Reply:
I am not getting How to add a dynamic payload. I tried but I did not get it. Please help me to solve it.